File: contains-duplicate/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  // let results = {};
  // for (let i = 0; i < nums.length; i++) {
  //   if (results[nums[i]] === undefined) {
  //     results[nums[i]] = 1;
  //   } else {
  //     return true;
  //   }
  // }
  // return false;
  let results = [...new Set(nums)]
  if (results.length === nums.length) {
    return false;
  }
  return true;
};

File: contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Arrays.sort(nums);
        HashSet<Integer> hash = new HashSet(20);
        for(int i = 0; i < nums.length; i++) {
            hash.add(nums[i]);
        }
        // System.out.println(hash.size());
        if(hash.size() != nums.length) {
            return true;
        }
        
        // List<Integer> listWithoutDuplicates = new ArrayList<>(new HashSet<>(nums));
        // System.out.println(Arrays.toString(nums));
        // for(int i = 1; i < nums.length; i++) {
        //     if(nums[i] == nums[i - 1]) {
        //         return true;
        //     }
        // }
        return false;
    }
}